spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/health/SpanCollectionHealthIndicator.java

```diff
@@ -0,0 +1,26 @@
+package org.springframework.cloud.sleuth.health;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.actuate.health.AbstractHealthIndicator;
+import org.springframework.boot.actuate.health.Health;
+
+/**
+ * Health indicator that shows the number of accepted / dropped spans.
+ *
+ * //TODO: should this be a health indicator (it's never down) or an Endpoint ?
```

Comment:
As an orthogonal note, we do work on exposing state for reactive, asynchronous (and even synchronous) flows in Spring 5 and any other Reactor supported projects. I plan to come back to you at some point but we are able to do this kind of crazy graphing : 
<img width="823" alt="screen shot 2016-01-10 at 8 25 18 pm" src="https://cloud.githubusercontent.com/assets/76718/12474310/1ab7a0d8-c012-11e5-93d1-cbcb70ec3c81.png">